service/auth.go

```diff
@@ -4,9 +4,56 @@
 package service
 
 import (
+	"encoding/json"
+	"fmt"
 	"net/http"
+
+	"github.com/mattermost/mattermost-server/v6/shared/mlog"
 )
 
 func (s *Service) wsAuthHandler(connID string, w http.ResponseWriter, r *http.Request) error {
+	clientID, authKey, ok := r.BasicAuth()
+	if !ok {
+		w.WriteHeader(http.StatusUnauthorized)
+		return fmt.Errorf("authentication failed: invalid auth header")
+	}
+
+	if err := s.auth.Authenticate(clientID, authKey); err != nil {
+		w.WriteHeader(http.StatusUnauthorized)
+		return fmt.Errorf("authentication failed: %w", err)
+	}
+
 	return nil
 }
+
+func (s *Service) registerClient(w http.ResponseWriter, req *http.Request) {
```

Comment:
The `/register` API endpoint doesn't currently require any authentication, which means clients can freely register themselves. This could potentially change or be configured if needed.